### PR TITLE
Feature/#1079 miner internals

### LINKED ISSUE #1079: Improve internal datatype that miner uses for node comparison
Make a `Vec2i` and move the miner to using that instead of `Point2D.Double`.

### PR DESCRIPTION
**Pull request by cltnschlosser**
Closes #1079

Commit log for more info

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java [hunk 1]
```diff
@@ -199,19 +199,19 @@ public void closeNextNode(final int rotation)
...
+        for (final Vec2i pos : nodeCenterList)
         {
             if (nodes.containsKey(pos))
             {
                 continue;
             }
-            final Node tempNodeToAdd = new Node(pos.getX(), pos.getY(), new Point2D.Double(tempNode.getX(), tempNode.getZ()));
+            final Node tempNodeToAdd = new Node(pos.getX(), pos.getZ(), new Vec2i(tempNode.getX(), tempNode.getZ()));
             final int randNumber = rand.nextInt(RANDOM_TYPES);
             tempNodeToAdd.setStyle(randNumber <= 1 ? TUNNEL : (randNumber == 2 ? BEND : CROSSROAD));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java]: MAJOR Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java [hunk 2]
```diff
@@ -1,13 +1,14 @@
 package com.minecolonies.coremod.entity.ai.citizen.miner;
 
 import com.minecolonies.coremod.colony.buildings.BuildingMiner;
+import com.minecolonies.coremod.util.Vec2i;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.common.util.Constants;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java]: MINOR Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 41 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingMiner.java#L473](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L473): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [BuildingMiner.java#L650](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L650): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L28](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L28): class "EntityAIStructureMiner" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L309](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L309): Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): The Cyclomatic Complexity of this method "getNextBlockInShaftToMine" is 32 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): The Cyclomatic Complexity of this method "getMinerProxy" is 23 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): Refactor this method to reduce its Cognitive Complexity from 56 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/448c055eb879f77d7650256e8634d417e119733e/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
Only missing codacy and sonar.
Else :shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1099/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java [hunk 3]
```diff
@@ -199,19 +198,33 @@ public void closeNextNode(final int rotation)
...
+            final Node tempNodeToAdd = new Node(pos.getX(), pos.getZ(), new Vec2i(tempNode.getX(), tempNode.getZ()));
+            tempNodeToAdd.setStyle(getRandomNodeType());
             nodes.put(pos, tempNodeToAdd);
             openNodes.add(tempNodeToAdd);
         }
-        nodes.get(new Point2D.Double(tempNode.getX(), tempNode.getZ())).setStatus(Node.NodeStatus.COMPLETED);
+        nodes.get(new Vec2i(tempNode.getX(), tempNode.getZ())).setStatus(Node.NodeStatus.COMPLETED);
+    }
+
+    private Node.NodeType getRandomNodeType()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java]: MINOR Make "getRandomNodeType" a "static" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2325)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingMiner.java#L473](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L473): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [BuildingMiner.java#L650](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L650): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L28](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L28): class "EntityAIStructureMiner" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L309](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L309): Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): The Cyclomatic Complexity of this method "getNextBlockInShaftToMine" is 32 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): The Cyclomatic Complexity of this method "getMinerProxy" is 23 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): Refactor this method to reduce its Cognitive Complexity from 56 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/4be28f249e0d67796b137184c5a4a50a3b2bec7d/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 39 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingMiner.java#L473](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L473): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [BuildingMiner.java#L650](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L650): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L28](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L28): class "EntityAIStructureMiner" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L309](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L309): Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): The Cyclomatic Complexity of this method "getNextBlockInShaftToMine" is 32 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): The Cyclomatic Complexity of this method "getMinerProxy" is 23 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L222](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L222): Refactor this method to reduce its Cognitive Complexity from 56 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/8fadcc370ecce59d097b40a398f88687a4e2b537/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/util/Vec2i.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package com.minecolonies.coremod.util;
```
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/util/Vec2i.java]: can we move that to api?

**Review by marvin-bitterlich — APPROVED** [review]
looking good, I would like to see the move ^^

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 38 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 28 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingMiner.java#L473](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L473): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [BuildingMiner.java#L650](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java#L650): Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L28](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L28): class "EntityAIStructureMiner" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L309](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L309): Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): The Cyclomatic Complexity of this method "getNextBlockInShaftToMine" is 32 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureMiner.java#L372](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java#L372): Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L221](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L221): The Cyclomatic Complexity of this method "getMinerProxy" is 23 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [WalkToProxy.java#L221](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java#L221): Refactor this method to reduce its Cognitive Complexity from 56 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/fa51942031fe1ecc8d98dd0a51878ca8a63dae3c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**